apisix/init.lua

```diff
@@ -868,6 +869,70 @@ local function healthcheck_passive(api_ctx)
 end
 
 
+function _M.status()
+    core.response.exit(200, "ok")
+end
+
+function _M.status_ready()
+    local_conf = core.config.local_conf()
+    local provider = core.table.try_read_attr(local_conf, "deployment", "role_traditional",
+                                              "config_provider")
+    if provider == "yaml" then
+        local status_shdict = ngx.shared.status_report_standalone
+        local pids = status_shdict:get_keys()
+        local resp = ""
+
+        for _, pid in pairs(pids) do
+            local ready = status_shdict:get(pid)
+            if not ready then
+                core.log.warn("worker pid: ", pid,
+                    " has not received configuration")
+                resp = resp .. "Worker pid: " .. pid .. " has not received configuration\n"
+                break
+            end
+        end
+
+        if resp and resp ~= "" then
+            core.response.exit(503, resp)
+            return
+        end
+
+        core.response.exit(200, "ok")
```

Comment:
We're better to return structured data (in JSON), and using text like this can't be interactive with downstream component. HTTP 503 isn't enough to convey enough detail.